msgpack/src/main/scala/fs2/data/msgpack/low/internal/ItemParser.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package msgpack
+package low
+package internal
+
+import fs2.data.msgpack.low.internal.Helpers._
+import fs2.data.msgpack.low.internal.FormatParsers._
+import scodec.bits._
+import scala.annotation.switch
+
+private[low] object ItemParser {
+
+  /** Parses a single item from the stream.
+    */
+  private def parseItem[F[_]](ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireOneByte(ctx).flatMap { res =>
+      val byte = res.result
+      val ctx = res.toContext
+
+      ((byte & 0xff): @switch) match {
+        case Headers.Nil       => Pull.pure(ctx.prepend(MsgpackItem.Nil))
+        case Headers.NeverUsed => Pull.raiseError(new MsgpackParsingException("Reserved value 0xc1 used"))
+        case Headers.False     => Pull.pure(ctx.prepend(MsgpackItem.False))
+        case Headers.True      => Pull.pure(ctx.prepend(MsgpackItem.True))
+        case Headers.Bin8      => parseBin(1, ctx)
+        case Headers.Bin16     => parseBin(2, ctx)
+        case Headers.Bin32     => parseBin(4, ctx)
+        case Headers.Ext8      => parsePlainExt(1, ctx)
+        case Headers.Ext16     => parsePlainExt(2, ctx)
+        case Headers.Ext32     => parsePlainExt(4, ctx)
+        case Headers.Float32   => parseSimpleType(MsgpackItem.Float32.apply)(4, ctx)
```

Comment:
It's a minor question of style, but I tend to prefer this syntax:

```suggestion
        case Headers.Float32   => parseSimpleType(MsgpackItem.Float32(_))(4, ctx)
```